src/rpc/subscription.h

```diff
@@ -0,0 +1,37 @@
+#ifndef EPIC_SUBSCRIPTION_H
+#define EPIC_SUBSCRIPTION_H
+
+#include "rpc_tools.h"
+#include <grpc++/grpc++.h>
+#include <rpc.grpc.pb.h>
+#include <rpc.pb.h>
+#include <string>
+#include <unordered_map>
+
+enum SubType { BLOCK = 1, TX = 2 };
+
+class Subscriber {
+public:
+    Subscriber(std::unique_ptr<rpc::Subscription::Stub>&& stub, uint8_t service_);
+    bool PushBlock(rpc::Vertex& vertex);
+    bool PushTx(rpc::Transaction& tx);
+    std::uint8_t service;
+
+private:
+    std::unique_ptr<rpc::Subscription::Stub> push_stub_;
+};
+
+
+class Publisher {
+public:
+    void AddNewSubscriber(std::string address, uint8_t service);
+    void DeleteSubscriber(std::string address);
+    void PushMsg(void* msg, SubType type);
+
+private:
+    std::unordered_map<std::string, Subscriber> subscribers_;
```

Comment:
I think `subscribers_` is accessed in multi-thread, so I suggest add mutex lock before access the variable.